dashboard/dashboard-front/src/components/jobs/JobsListView.vue

```diff
@@ -0,0 +1,266 @@
+<script setup lang="ts">
+import { computed, nextTick, onMounted, ref, watch, type Ref } from 'vue'
+import { QTree } from 'quasar'
+import { apiClient } from '@/services/ApiClient'
+import { toastService } from '@/services/ToastService'
+import { type JobData } from '@/utils/api-schema'
+import JobDetails from '@/components/jobs/JobDetails.vue'
+import JobStatus from '@/components/jobs/JobStatus.vue'
+
+enum JobOrder {
+    ByLatestFamily,
+    ByLatestJob,
+    ByName,
+}
+
+const jobsData: Ref<JobData[]> = ref([])
+const jobsQTreeRef: Ref<QTree | null> = ref(null)
+const splitterModel: Ref<number> = ref(30)
+const treeFilter: Ref<string> = ref('')
+const jobsByKey: Ref<Map<string, JobData>> = ref(new Map())
+const jobsTree: Ref<any[]> = ref([])
+const currentJob: Ref<JobData | null> = ref(null)
+const jobsCount: Ref<number> = computed(() => jobsData.value?.length || 0)
+const selectedNodeKey: Ref<string | null> = ref(null)
+const jobOrder: Ref<JobOrder> = ref(JobOrder.ByName)
+const loadingTree = ref(true)
+
+function fetchJobs() {
+    loadingTree.value = true
+    apiClient.get(`/api/v1/job`).then(response => {
+        jobsData.value = response.data
+    }).catch(err => {
+        toastService.showErrorDetails(`Failed to fetch the jobs`, err)
+    }).finally(() => {
+        loadingTree.value = false
+    })
+}
+
+function populateJobsData() {
+    const sortedJobs: JobData[] = [...jobsData.value]
+    if (jobOrder.value === JobOrder.ByLatestFamily || jobOrder.value === JobOrder.ByLatestJob) {
+        sortedJobs.sort((a, b) => {
+            return b.update_time - a.update_time
+        })
+    } else if (jobOrder.value === JobOrder.ByName) {
+        sortedJobs.sort((a, b) => {
+            return a.name.toLowerCase().localeCompare(b.name.toLowerCase())
+                || compareVersions(b.version, a.version)
+        })
+    }
```

Comment:
You could use the immutable method `toSorted` here so you don't make a copy of the array before sorting it. https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/toSorted

But this one is also fine. Either way, you have to save it to the separate variable.

Reply:
I've tried to use `toSorted` but Typescript says there's no such property on an array. Perhaps it's because it's not fully supported yet https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/toSorted#browser_compatibility.